### PR TITLE
feat(Scripts/Commands): Morph Mount

### PR DESCRIPTION
**Pull request by Gultask**
Something I was toying around with.
There's also the Mount() method but I prefer using displayids instead

## Changes Proposed:

This PR proposes changes to:
-  [X] Core (units, players, creatures, game systems).
-  [ ] Scripts (bosses, spell scripts, creature scripts).
-  [ ] Database (SAI, creatures, etc).

## Issues Addressed:

- Closes 

## SOURCE:

The changes have been validated through:
- [ ] Live research (checked on live servers, e.g Classic WotLK, Retail, etc.)
- [ ] Sniffs (remember to share them with the open source community!)
- [ ] Video evidence, knowledge databases or other public sources (e.g forums, Wowhead, etc.)
- [ ] The changes promoted by this pull request come partially or entirely from another project (cherry-pick). **Cherry-picks must be committed using the proper --author tag in order to be accepted, thus crediting the original authors, unless otherwise unable to be found**

## Tests Performed:

This PR has been:
- [X] Tested in-game by the author.
- [ ] Tested in-game by other community members/someone else other than the author/has been live on production servers.
- [ ] This pull request requires further testing and may have edge cases to be tested.

## How to Test the Changes:

- [ ] This pull request can be tested by following the reproduction steps provided in the linked issue
- [ ] This pull request requires further testing. Provide steps to test your changes. If it requires any specific setup e.g multiple players please specify it as well.

1. mount up
2. .morph mount 92
3.

## Known Issues and TODO List:

- [ ]
- [ ]

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

**Comment by heyitsbench** [comment]
Also closes a feature request issue somewhere.

**Comment by Gultask** [comment]
🧐

**Comment by heyitsbench** [comment]
https://github.com/azerothcore/azerothcore-wotlk/issues/1141

> Funny command for GM to mount any creature

I misremembered. 😛

**Comment by Nyeriah** [comment]
I could be mistaken but it’s not displaying any message on error?

**Comment by Gultask** [comment]
Not sure I mostly copied from the morph target code so I imagined it spits out an error if it returns false.

**Comment by Gultask** [comment]
image

if player is not mounted, and if invalid value

File: data/sql/updates/pending_db_world/rev_1722817302187444800.sql [hunk 1]
```diff
@@ -0,0 +1,4 @@
+--
+DELETE FROM `command` WHERE `name` = 'morph mount';
+INSERT INTO `command` (`name`, `security`, `help`) VALUES
+('morph mount', 1, 'Syntax: .morph mount #displayid - Change the selected target\'s mount\'s model id to #displayid.');
```
Comment (heyitsbench) [inline data/sql/updates/pending_db_world/rev_1722817302187444800.sql:4]:
```suggestion
('morph mount', 1, 'Syntax: .morph mount #displayid - Change the selected target\'s mount\'s model ID to #displayid.');
```

Comment (Gultask) [reply data/sql/updates/pending_db_world/rev_1722817302187444800.sql:4]: shut up

File: src/server/scripts/Commands/cs_modify.cpp [hunk 2]
```diff
@@ -868,6 +869,23 @@ class modify_commandscript : public CommandScript
         return true;
     }
 
+    static bool HandleMorphMountCommand(ChatHandler* handler, uint32 displayID)
+    {
+        Unit* target = handler->getSelectedUnit();
```
Comment (Nyeriah) [inline src/server/scripts/Commands/cs_modify.cpp:874]: getSelectedUnitOrSelf()

Comment (Nyeriah) [reply src/server/scripts/Commands/cs_modify.cpp:874]: then you won't need 876 and 877

**Review by Nyeriah — APPROVED** [review]

**Pull request merged**